### PR TITLE
[#388] Sync `files.text` with the OCRd text from DocumentCloud

### PR DESCRIPTION
**Pull request by vitorbaptista**
This processor updates the column `files.text` in our database with the
contents of the full text extraction from DocumentCloud.

Fixes opentrials/opentrials#388

### CODE REVIEW

File: processors/sync_text_from_documentcloud/processor.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+def _get_text(client, doc_id):
+    doc = None
+    text = None
+
+    try:
+        doc = client.documents.get(doc_id)
+    except Exception as e:
+        is_http_error_403 = (getattr(e, 'code', None) == 403)
+        if is_http_error_403:
+            raise e
```
Comment (victor-nitu) [inline processors/sync_text_from_documentcloud/processor.py:38]: Any idea why is this happening?

File: processors/sync_text_from_documentcloud/processor.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        logger.debug(msg)
+
+    return text
+
+
+def _documentcloud_client(conf):
+    username = conf['DOCUMENTCLOUD_USERNAME']
+    password = conf['DOCUMENTCLOUD_PASSWORD']
+
+    return documentcloud.DocumentCloud(username, password)
```
Comment (victor-nitu) [inline processors/sync_text_from_documentcloud/processor.py:56]: ouch, the alternative import looks even more redundant :D 
 `from documentcloud import DocumentCloud as documentcloud`

File: processors/sync_text_from_documentcloud/processor.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import documentcloud
+import logging
+from .. import base
+logger = logging.getLogger(__name__)
+
+
+def process(conf, conn):
+    query = (
+        'SELECT id, documentcloud_id FROM files'
+        ' WHERE documentcloud_id IS NOT NULL'
```
Comment (victor-nitu) [inline processors/sync_text_from_documentcloud/processor.py]: I'd go for a triple quoted multiline string on this one, just for the ease of maintainability. But it really doesn't matter that much in this case, up to you.

**Comment by victor-nitu** [comment]
All seems to be good, ping me if you feel like I should run it. If not, I'd merge it. If yes, you need to share some `.env` bits :)

File: processors/sync_text_from_documentcloud/processor.py (hunk 1 above)
Comment (vitorbaptista) [reply processors/sync_text_from_documentcloud/processor.py:38]: It happens when the password is wrong or you don't have permissions. It shouldn't happen normally, catching here just to, if it does, we stop processing altogether.

**Pull request merged**